pkg/networkservice/mechanisms/srv6/client.go

```diff
@@ -0,0 +1,147 @@
+
+	"github.com/networkservicemesh/api/pkg/api/connection"
+	"github.com/networkservicemesh/api/pkg/api/connection/mechanisms/srv6"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/vppagent"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+type srv6Client struct{}
+
+// NewClient provides a NetworkServiceClient chain elements that support the srv6 Mechanism
+func NewClient() networkservice.NetworkServiceClient {
+	return &srv6Client{}
+}
+
+func (v *srv6Client) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*connection.Connection, error) {
+	if err := v.appendInterfaceConfig(ctx, request.GetConnection(), true); err != nil {
+		return nil, err
+	}
+	return next.Client(ctx).Request(ctx, request, opts...)
+}
+
+func (v *srv6Client) Close(ctx context.Context, conn *connection.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
+	if err := v.appendInterfaceConfig(ctx, conn, false); err != nil {
+		return nil, err
+	}
+	return next.Client(ctx).Close(ctx, conn, opts...)
+}
+
+func (v *srv6Client) appendInterfaceConfig(ctx context.Context, conn *connection.Connection, connect bool) error {
+	conf := vppagent.Config(ctx)
+	if mechanism := srv6.ToMechanism(conn.GetMechanism()); mechanism != nil {
+		dstHostLocalSID, err := mechanism.SrcHostLocalSID()
+		if err != nil { return err }
+		hardwareAddress, err := mechanism.SrcHardwareAddress()
+		if err != nil { return err }
+		srcBSID, err := mechanism.SrcBSID()
+		if err != nil { return err }
+		srcLocalSID, err := mechanism.SrcLocalSID()
+		if err != nil { return err }
+		dstLocalSID, err := mechanism.DstLocalSID()
+		if err != nil { return err }
+
+		var localIfaceName string
+		if ifaces := conf.GetVppConfig().GetInterfaces(); len(ifaces) == 1 {
+			localIfaceName = ifaces[0].Name
+		} else {
+			return errors.Errorf("failed to choose local interface for srv6 mechanism: %v", ifaces)
+		}
+
+		conf.GetVppConfig().Srv6Localsids = []*vpp_srv6.LocalSID{
+			{
+				Sid: srcLocalSID,
+				EndFunction: &vpp_srv6.LocalSID_EndFunction_DX2{
+					EndFunction_DX2: &vpp_srv6.LocalSID_EndDX2{
+						VlanTag:           math.MaxUint32,
+						OutgoingInterface: localIfaceName,
+					},
+				},
+			},
+		}
+		conf.GetVppConfig().Srv6Policies = []*vpp_srv6.Policy{
+			{
+				Bsid: srcBSID,
+				SegmentLists: []*vpp_srv6.Policy_SegmentList{
+					{
+						Segments: []string{
+							dstHostLocalSID,
+							dstLocalSID,
+						},
+						Weight: 0,
+					},
+				},
+				SrhEncapsulation: true,
+			},
+		}
+
+		conf.GetVppConfig().Srv6Steerings = []*vpp_srv6.Steering{
+			{
+				Name: conn.GetId(),
+				PolicyRef: &vpp_srv6.Steering_PolicyBsid{
+					PolicyBsid: srcBSID,
+				},
+				Traffic: &vpp_srv6.Steering_L2Traffic_{
+					L2Traffic: &vpp_srv6.Steering_L2Traffic{
+						InterfaceName: localIfaceName,
+					},
+				},
+			},
+		}
+
+		if connect {
```

Comment:
Oh, this is interesting... so if we try to delete the VRF entry here on 'Close' it causes issues?

Reply:
Yes. VRF with id=math.MaxUint32 is valid for all SRv6 policies. I am sending it with each new connection, because vpp can delete it by itself if last srv6 policy was deleted.

Reply:
Super interesting!